cfn.template

```diff
@@ -147,39 +147,51 @@ Resources:
+                - |
+                  if [ -n "$template" ]; then
+                      templates="$template"
+                  fi
+
                 - |
-                  cat >> publish.sh << 'EOL'
+                  cat <<"EOL" >publish.sh
                   #!/bin/bash
                   set -e
+                  set -x
+                  # Loop through regions
                   for region in $regions; do
                     echo Publishing to "$region"...
                     export AWS_DEFAULT_REGION="$region"
                     export BUCKET="${bucket_name_prefix}-${region}"
 
                     # Make the bucket if it doesn't exist
-                    aws s3api head-bucket --bucket "$BUCKET" 2>/dev/null || aws s3 mb "s3://$BUCKET" &>/dev/null
+                    aws s3api head-bucket --bucket "$BUCKET" 2>/dev/null || aws s3 mb "s3://$BUCKET"
 
                     # Give the bucket time to existify
                     aws s3api wait bucket-exists --bucket "$BUCKET"
 
-                    # Make sure it can host websites
-                    aws s3api put-bucket-website --bucket "$BUCKET" --website-configuration '{"IndexDocument": {"Suffix": "index.html"}}'
-
                     # Clear existing files
                     echo Clearing existing files for version: "$version"
                     aws s3 rm --recursive "s3://${BUCKET}/${project}/${version}"
 
-                    # Package the template
-                    aws cloudformation package --template-file "$template" --s3-bucket "$BUCKET" --s3-prefix "${project}/${version}" --output-template-file packaged.template
+                    # Loop through templates
+                    for template in $templates; do
+                      # Package the template
+                      aws cloudformation package --template-file "$template" --s3-bucket "$BUCKET" --s3-prefix "${project}/${version}" --output-template-file packaged.template
 
-                    # Copy it to the bucket
-                    aws s3 cp --acl "$acl" packaged.template "s3://${BUCKET}/${project}/${version}/main.template"
+                      # Define paths
+                      path="${project}/${version}/$(basename $template)"
 
-                    url="https://s3.amazonaws.com/${BUCKET}/${project}/${version}/main.template"
+                      # Copy it to the bucket
+                      aws s3 cp --acl "$acl" packaged.template "s3://${BUCKET}/${path}"
 
-                    # Add an index
-                    echo "<!DOCTYPE html><html lang=\"en\"><body><p><a href=\"https://console.aws.amazon.com/cloudformation/home?region=${region}#/stacks/new?stackName=${project}&templateURL=${url}\"><img alt=\"Launch Stack\" src=\"https://s3.amazonaws.com/cloudformation-examples/cloudformation-launch-stack.png\" /></a></p></body></html>" >index.html
-                    aws s3 cp --acl "$acl" index.html "s3://${BUCKET}/${project}/${version}/index.html"
+                      # Sync to latest
```

Comment:
shouldn't the sync still remain outside the for loop since it's recursive anyways. If we only sync here, the `extra_files` won't get synced to `latest`

Reply:
Rolled back based on our OOB conversation :)